Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -0,0 +1,137 @@
+        /// </summary>
+        public string FindRichestPerson()
+        {
+            Account richestBalanceAccount;
+            try
+            {
+                richestBalanceAccount = _accounts[0];
+                for (int i = 1; i < _accounts.Length; i++)
+                {
+                    if (_accounts[i].GetCurrentBalance() > richestBalanceAccount.GetCurrentBalance())
+                    {
+                        richestBalanceAccount = _accounts[i];
+                    }
+                }
+                return _accounts.Length > _arrayBreak && AccountOps.AreBalancesEqual(_accounts)
+               ? $"{StringOps.FormatAndCommas(_accounts)} {Messages.AreTheRichestPeople}. {StringOps.FormatCurrency(richestBalanceAccount.GetCurrentBalance(), _currency)}."
+               : $"{richestBalanceAccount.GetName()} {Messages.IsTheRichestPerson}. {StringOps.FormatCurrency(richestBalanceAccount.GetCurrentBalance(), _currency)}.";
+            }
+            catch (Exception ex) when (ex is NullReferenceException || ex is IndexOutOfRangeException)
+            {
+                return Messages.InvalidMessage;
+            }
+            return _accounts.Length > _arrayBreak && AccountOps.AreBalancesEqual(_accounts)
+              ? $"{StringOps.FormatAndCommas(_accounts)} {Messages.AreTheRichestPeople}. {StringOps.FormatCurrency(richestBalanceAccount.GetCurrentBalance(), _currency)}."
+              : $"{richestBalanceAccount.GetName()} {Messages.IsTheRichestPerson}. {StringOps.FormatCurrency(richestBalanceAccount.GetCurrentBalance(), _currency)}.";
+        }
+
+        /// <summary>
+        /// Return name and current money of the most poor person.
+        /// </summary>
+        public string FindMostPoorPerson()
+        {
+            Account poorestBalanceAccount;
+            try
+            {
+                poorestBalanceAccount = _accounts[0];
+                for (int i = 1; i < _accounts.Length; i++)
+                {
+                    if (_accounts[i].GetCurrentBalance() < poorestBalanceAccount.GetCurrentBalance())
+                    {
+                        poorestBalanceAccount = _accounts[i];
+                    }
+                }
+            }
+            catch (Exception ex) when (ex is NullReferenceException || ex is IndexOutOfRangeException)
+            {
+                return Messages.InvalidMessage;
+            }
+            return _accounts.Length > _arrayBreak && AccountOps.AreBalancesEqual(_accounts)
+                ? $"{StringOps.FormatAndCommas(_accounts)} {Messages.HaveTheLeastMoney}. {StringOps.FormatCurrency(poorestBalanceAccount.GetCurrentBalance(), _currency)}."
+                : $"{poorestBalanceAccount.GetName()} {Messages.HasTheLeastMoney}. {StringOps.FormatCurrency(poorestBalanceAccount.GetCurrentBalance(), _currency)}.";
+            }
+            catch (Exception ex) when (ex is NullReferenceException || ex is IndexOutOfRangeException)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Make 'when' a static method.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5054323)